File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> st = new Stack<>();
        
        for(int i = 0; i < s.length(); i++){
            char c = s.charAt(i);
            if(c == '('){
                st.push(c);
            }
            else if(c == ')'){
                if(st.isEmpty()){
                    return false;
                }
                st.pop();
            }
        }
        if(st.isEmpty()){
            return answer;
        }
        else{
            return false;
        }
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for char in s:
        if not stack:
            stack.append(char)
        else:
            if char == '(':
                stack.append(char)
            else:
                stack.pop()
        # print(stack)
    if not stack:
        return True
    else:
        return False